subprojects/docs/src/docs/release/notes.md

```diff
@@ -18,6 +18,12 @@ TBD - This release builds on the `Provider` concept added in Gradle 4.0 to add c
 - Added a `ProjectLayout` service that allows input and output locations to be defined relative to the project's project directory and build directory. 
 - `Project.file()` and `Project.files()` can resolve `Provider` instances to `File` and `FileCollection` instances.
 
+### Features for easier plugin authoring
+
+While it is easy for a plugin author to extend the Gradle DSL to add top level blocks to the DSL using project extensions, in previous versions of Gradle it was awkward to create a deeply nested DSL inside these top level blocks, often requiring the use of internal Gradle APIs.
```

Comment:
Should we maybe mention `Instantiator` explicitly so users know which internal API we are talking about? I have the gut feeling that some users just use the class without knowing what it does exactly because they found it somewhere in the Gradle core code base.